File: java_2_gb/OOP/src/main/java/org/example/seminar5/cw/model/Drinkables.java
package org.example.seminar5.cw.model;

import java.math.BigDecimal;

public class Drinkables extends Product {
    private Double volume;
    public Drinkables(String name, BigDecimal price, Double volume) {
        super(name, price);
        this.volume = volume;
    }
    public Double getVolume() {
        return volume;
    }
    @Override
    public String toString() {
        return super.toString() + "объем = " + volume + " ";
    }
    public void setVolume(Double volume) {
        if (volume > 0) this.volume = volume;
        else System.out.println("Объем напитка не может быть равен или меньше нуля!");
    }
}

File: java_2_gb/finalTask/Model/Goods/Drinkables.py
class Drinkables(Product):
    def __init__(self, name, price, volume):
        super().__init__(name, price)
        self._volume = volume

    @property
    def volume(self):
        return self._volume

    @volume.setter
    def volume(self, new_volume):
        self._volume = new_volume

    def __str__(self) -> str:
        return f"{super().__str__()}, объем: {self.volume}"